### PR TITLE
Simplify config and port setup

### PR DESCRIPTION
**Pull request by fdegier**

### CODE REVIEW

**Comment by fdegier** [comment]
@moyix @Frederisk this is a follow up from the discussion we had in https://github.com/moyix/fauxpilot/pull/66

I think this PR superseded #66 if it resolved the discussions we still had left.

**Comment by moyix** [comment]
This looks good to me, but my GPUs are currently occupied on a long-running task so I can't test immediately. Will do so once I have a free GPU and can run a quick smoke test! :)

**Comment by fdegier** [comment]
@moyix thanks, I tried testing different scenarios but its quite difficult. I tried setting up some automated tests but with my Apple Silicon there is not a lot that I can do due to dependency issues and not having a GPU :D

**Review by moyix — CHANGES_REQUESTED** [review]
I noticed two issues in setup.sh with this, but otherwise it looks good!

File: setup.sh [hunk 1]
```diff
@@ -44,46 +50,68 @@ case $MODEL_NUM in
...
-echo "MODEL_DIR=${MODEL_DIR}" >> config.env
+# Write .env
+echo "MODEL=${MODEL}" > .env
+echo "NUM_GPUS=${NUM_GPUS}" >> .env
+echo "MODEL_DIR=${MODEL_DIR}/${MODEL}-${NUM_GPUS}gpu" >> .env
+echo "API_EXTERNAL_PORT=${API_EXTERNAL_PORT}" >> .env
+echo "TRITON_HOST=${TRITON_HOST}" >> .env
+echo "TRITON_PORT=${TRITON_PORT}" >> .env
+GPUS="$(seq -s ','  -t '\n' "${NUM_GPUS}")"
+echo "GPUS=${GPUS%?}" >> .env
```
[the remark this answers is not in the document]
Comment (fdegier) [reply setup.sh:81]: Thanks for the review! Somewhere the 0 zero got lost in my trial and error to get the comma's. 

Currently in main there is `export GPUS=$(seq 0 $(( NUM_GPUS - 1 )) | paste -sd ',')` which doesn't work which you now solved by adding the `-`. That works, I've added it to a new commit.

File: launch.sh [hunk 2]
```diff
@@ -1,19 +1,11 @@
...
-export GPUS=$(seq 0 $(( NUM_GPUS - 1 )) | paste -sd ',')
+source .env
 
 # On newer versions, docker-compose is docker compose
-if command -v docker-compose > /dev/null; then
-    docker compose up
-else
-    docker-compose up
-fi
+docker compose up -d --remove-orphans || docker-compose up -d --remove-orphans
```
Comment (fdegier) [inline launch.sh:11]: I was having issues with `orphan` containers, as well as the `command -v docker-compose > /dev/null; then` resulting in `docker compose` leading to `docker: 'compose' is not a docker command.`

**Pull request merged**